lib/Cmts.ml

```diff
@@ -626,25 +626,9 @@ let remaining_before t loc = Location.Multimap.find_multi t.cmts_before loc
 let remaining_locs t = Location.Set.to_list t.remaining
 
 let diff (conf : Conf.t) x y =
-  let norm z =
-    let norm_non_code {Cmt.txt; _} = Normalize.comment txt in
-    let f z =
-      match Cmt.txt z with
-      | "" | "$" -> norm_non_code z
-      | str ->
-          if Char.equal str.[0] '$' then
-            let chars_removed =
-              if Char.equal str.[String.length str - 1] '$' then 2 else 1
-            in
-            let len = String.length str - chars_removed in
-            let str = String.sub ~pos:1 ~len str in
-            try
-              Migrate_ast.Parse.implementation (Lexing.from_string str)
-              |> Normalize.impl conf
-              |> Caml.Format.asprintf "%a" Printast.implementation
-            with _ -> norm_non_code z
-          else norm_non_code z
-    in
-    Set.of_list (module String) (List.map ~f z)
+  let norm_cmt =
+    Cmt.normalized_string ~normalize_comment:Normalize.comment
+      ~normalize_impl:(Normalize.impl conf)
```

Comment:
I wonder if we should instead:
- move `Normalize.comment` into `Cmt` (like `Cmt.basic_normalize` that does not need to be exported)
- rename `Cmt.normalized_string` to `Cmt.normalize`, that would call `basic_normalize` instead of `normalize_comment`

Reply:
We can't have a `Cmt` -> `Normalize` dependency so `normalize_impl` will have to be passed in, does that sound OK?

Reply:
Maybe I missed something but `Normalize.comment` does not pull additional Normalize dependencies.

Reply:
OK I though you meant doing this step in addition of this PR, not instead of. I'll try something.